### PR TITLE
build requirements with python3.8

### PR DESCRIPTION
**Pull request by ryanpetrello**

### CODE REVIEW

File: requirements/requirements.txt [hunk 1]
```diff
@@ -148,22 +148,12 @@ hyperlink==20.0.1
     # via
     #   autobahn
     #   twisted
-idna-ssl==1.1.0
```
Comment (ryanpetrello) [inline requirements/requirements.txt:151]: This is unnecessary in Python3.7+

https://github.com/aio-libs/aiohttp/issues/949
https://github.com/python/cpython/pull/3010

File: requirements/requirements.txt [hunk 2]
```diff
@@ -148,22 +148,12 @@ hyperlink==20.0.1
...
-    # via aiohttp
 idna==2.9
     # via
     #   hyperlink
-    #   idna-ssl
     #   requests
     #   twisted
     #   yarl
-importlib-metadata==1.5.0
```
Comment (ryanpetrello) [inline requirements/requirements.txt:160]: These are unnecessary in Python3.8

File: requirements/requirements.txt [hunk 3]
```diff
@@ -391,8 +381,6 @@ twisted[tls]==20.3.0
     #   daphne
 txaio==20.12.1
     # via autobahn
-typing-extensions==3.7.4.1
```
Comment (ryanpetrello) [inline requirements/requirements.txt:394]: `typing` is no longer provisional in 3.8, this polyfill library is unnecessary in 3.8

File: requirements/requirements.txt [hunk 4]
```diff
@@ -409,10 +397,6 @@ xmlsec==1.3.3
     # via python3-saml
 yarl==1.4.2
     # via aiohttp
-zipp==3.1.0
```
Comment (ryanpetrello) [inline requirements/requirements.txt:412]: This isn't needed because its parent dependencies are gone

File: requirements/requirements.txt (hunk 2 above)
Comment (ryanpetrello) [reply requirements/requirements.txt:160]:
```shell
bash-4.4$ python3.8 -c 'from importlib import metadata'
bash-4.4$ python3.6 -c 'from importlib import metadata'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: cannot import name 'metadata'

bash-4.4$ python3.8 -c 'from importlib import resources'
bash-4.4$ python3.6 -c 'from importlib import resources'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: cannot import name 'resources'
```

File: requirements/requirements.txt (hunk 3 above)
Comment (ryanpetrello) [reply requirements/requirements.txt:394]:
```
bash-4.4$ python3.6 -c 'from typing import Protocol'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: cannot import name 'Protocol'
bash-4.4$ python3.8 -c 'from typing import Protocol'
```

**Review by Spredzy — APPROVED** [review]

**Pull request merged**